test/box-luatest/lua_memory_test.lua

```diff
@@ -0,0 +1,42 @@
+local server = require('luatest.server')
+local t = require('luatest')
+local g = t.group('lua_memory setting test')
+
+local MB = 1024 * 1024
+local LUA_MIN_MEMORY = 256 * MB
+
+g.before_all(function(g)
+    g.server = server:new{
+        alias = 'default',
+        box_cfg = {lua_memory = LUA_MIN_MEMORY}
+    }
+    g.server:start()
+end)
+
+g.after_all(function(g)
+    g.server:drop()
+end)
+
+g.test_lua_memory_setting = function(g)
+    g.server:exec(function()
+        local MB = 1024 * 1024
+        local LUA_MIN_MEMORY = 256 * MB
+
+        t.assert_equals(box.cfg.lua_memory, LUA_MIN_MEMORY)
+
+        -- Increasing the limit should be ok.
+        box.cfg{lua_memory = LUA_MIN_MEMORY + 1}
+        t.assert_equals(box.cfg.lua_memory, LUA_MIN_MEMORY + 1)
+
+        local err_decreasing = 'cannot decrease lua allocator available '..
+            'memory at runtime'
+        -- Decreasing shouldn't be allowed.
+        t.assert_error_msg_equals(err_decreasing, box.cfg,
+            {lua_memory = LUA_MIN_MEMORY})
+
+        local err_min = 'cannot limit lua memory with values less than 256MB'
+        -- Setting the limit to values less than should be disabled.
```

Comment:
Nit: Comment length is more than 66 symbols.

Reply:
Made it shorter.

Reply:
Fixed.